src/app/containers/PageHandlers/withOptimizelyProvider/index.jsx

```diff
@@ -1,15 +1,36 @@
 import React, { useContext } from 'react';
 import { createInstance, OptimizelyProvider } from '@optimizely/react-sdk';
+import { v4 as uuid } from 'uuid';
+import Cookie from 'js-cookie';
 import { ServiceContext } from '#contexts/ServiceContext';
-import { getAtUserId } from '#lib/analyticsUtils';
+import onClient from '#lib/utilities/onClient';
+import isOperaProxy from '#lib/utilities/isOperaProxy';
+
+const getCknsMvt = () => {
+  // Users accessing the site on opera "extreme data saving mode" have the pages rendered by an intermediate service
+  // Attempting to track these users is just tracking that proxy, causing all opera mini visitors to have the same id
+  if (!onClient() || isOperaProxy()) return null;
+
+  const cookieName = 'ckns_mvt';
+  const cookieValue = Cookie.get(cookieName);
+  const expires = 365; // expires in 12 Months
+
+  if (!cookieValue) {
+    const cookieUuid = uuid();
+    Cookie.set(cookieName, uuid(), { expires, path: '/' });
+    return cookieUuid;
+  }
+
+  return cookieValue;
+};
 
 const optimizely = createInstance({
   sdkKey: process.env.SIMORGH_OPTIMIZELY_SDK_KEY,
   eventBatchSize: 10,
   eventFlushInterval: 1000,
 });
 
-const withOptimizelyProvider = (Component, noUserId = false) => {
+const withOptimizelyProvider = (Component, hasUserId = true) => {
```

Comment:
another clean code best practice is to avoid [passing booleans as arguments](https://medium.com/@amlcurran/clean-code-the-curse-of-a-boolean-parameter-c237a830b7a3)

Can we not determine is there is a userId inside of this function?

Reply:
We have refactored this so that instead of setting `hasUserId` as above, we instead check to see if storybook is loaded on the page or if Optimizely is on live in order to set the userId to null - seems to be working ok on dev and storybook!